quickfixj-perf-test/README.md

```diff
@@ -0,0 +1,53 @@
+# QuickFIX/J Performance test
+
+This is a [JMH](https://github.com/openjdk/jmh) benchmark module for QuickFIX/J FIX protocol implementation. 
+
+## How to run
+
+### Using your favorite IDE
+
+Performance regression classes can be individually run using your favorite IDE.
+
+### Creating executable jar for the performance testing
+
+Build executable jar using following maven command
+
+```
+$ mvn clean package    
+```
+
+Use following command to run complete set of performance regression test cases
+
+```
+$ java -jar target/quickfixj-perf-test.jar
+```
+
+You can list available performance benchmarks using `-l` option
+```
+$ java -jar target/quickfixj-perf-test.jar -l
+```
+
+You can run individual benchmarks by providing the class name or benchmark method name
+```
+$ java -jar target/quickfixj-perf-test.jar MessageCrackerPerfTest.crack
+
+$ java -jar target/quickfixj-perf-test.jar MessageCrackerPerfTest
+```
+
+You can change the time unit used in the benchmark test using `-tu` option
+Following command is an example of using micro second for describing benchmark test results.
+```
+$ java -jar target/quickfixj-perf-test.jar MessageCrackerPerfTest -tu us
+```
+
+For more available options use `-h` option
+
+```
+$ java -jar target/quickfixj-perf-test.jar -h
+```
+
+#### Guideline for future performance enhancements
+
+1. Check if there is already benchmark available for the code you are planning to optimize.
+2. If there is no benchmark code, first create a benchmark and make pull request.
+3. Your performance improvement can use the benchmark code as the baseline for comparison.
```

Comment:
Just a quick question:
So a user wants to change some code and supplies a PR with a benchmark. Then that user should add the current code to the `baseline()` method and the changed code in an additional method (as you have done in your benchmarks below)?
What happens with the benchmark after the PR has been merged?

Reply:
hi @chrjohn 

I think I use the baseline() work incorrectly here. baseline is always empty method. 
When someone make code changes, its difficult expect that previous code will still be there to create a baseline.
Rather what I suggest is comparison between two branches. Say master and `a-perf-improvement` branch.

I have updated the README.md to make this more clear. Kindly have a look.